app/views/shared/_article_fields_price.html.haml

```diff
@@ -17,18 +21,62 @@
       = Article.human_attribute_name(:fc_price)
     .controls.control-text#article_fc_price
       = number_to_currency(f.object.fc_price) rescue nil
+.fold-line
+  = f.input :supplier_price do
+    .input-prepend
+      %span.add-on= t 'number.currency.format.unit'
+      = f.input_field :supplier_price, class: 'input-mini'
+
 
 -# do this inline, since it's being used in ajax forms only
 - field = f.object.class.model_name.to_s.underscore
 :javascript
   var form = $('#article_fc_price').closest('form');
-  $('##{field}_price, ##{field}_tax, ##{field}_deposit', form).on('change keyup', function() {
+  function updatePrice(){
     var price = parseFloat($('##{field}_price', form).val());
     var tax = parseFloat($('##{field}_tax', form).val());
     var deposit = parseFloat($('##{field}_deposit', form).val());
     // Article#gross_price and Article#fc_price
     var gross_price = (price + deposit) * (tax / 100 + 1);
     var fc_price = gross_price  * (#{FoodsoftConfig[:price_markup].to_f} / 100 + 1);
     $('#article_fc_price').html($.isNumeric(fc_price) ? I18n.l("currency", fc_price) : '&#133;');
-  });
+    checkAndNotifyPriceProblem();
+  }
+
+  function computeSplitPrice(){
+    var units = parseFloat($('##{field}_unit_quantity').val());
+    var supplierPrice = parseFloat($('##{field}_supplier_price', form).val());
+    var price = ((supplierPrice/units)*100);
+    price = Math.ceil(price);
+    price = price/100;
+    return price;
+  }
+
+  function updateSplitPrice(){
+    var price = computeSplitPrice();
+    var oldPrice = parseFloat($('##{field}_price', form).val());
+    if (oldPrice == price){
+      window.alert(      [
+      'this button lets you quickly recalculate the split price, but the',
+      'price is still the same!  price = (supplier price/unit)',
+      'did you change unit or supplier price?',
+      ].join(' '));
```

Comment:
Candidate for i18n-js (see also `config/i18n-js.yml` for which strings are included in the js).

Reply:
agree and done